source/plugins/go/src/network_flow_logs.go

```diff
@@ -0,0 +1,380 @@
+		}
+	}()
+
+	flow, ok := record["flow"].(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("'flow' field not found or is not a map")
+	}
+	// TimeGenerated
+	dataMap["TimeGenerated"] = extractString(flow, "time")
+	// UUID
+	if uuidVal := extractString(record, "UUID"); uuidVal != "" {
+		dataMap["UUID"] = uuidVal
+	} else {
+		dataMap["UUID"] = uuid.New().String()
+	}
+	// Verdict and DropReason
+	dataMap["Verdict"] = extractString(flow, "verdict")
+	dataMap["DropReason"] = extractString(flow, "drop_reason_desc")
+	// IP
+	if ip, ok := flow["IP"].(map[string]interface{}); ok {
+		dataMap["IP"] = serializeToJSON(ip)
+	}
+	// Layer4
+	if l4, ok := flow["l4"].(map[string]interface{}); ok {
+		dataMap["Layer4"] = serializeToJSON(l4)
+	}
+	// Source details
+	if source, ok := flow["source"].(map[string]interface{}); ok {
+		dataMap["SourceIdentity"] = safeToInt(source["identity"])
+		dataMap["SourceClusterName"] = extractString(source, "cluster_name")
+		dataMap["SourceNamespace"] = extractString(source, "namespace")
+		dataMap["SourcePodName"] = extractString(source, "pod_name")
+		if workloads, ok := source["workloads"].([]interface{}); ok {
+			dataMap["SourceWorkloads"] = serializeToJSON(workloads)
+		}
+	}
+	// Destination details
+	if dest, ok := flow["destination"].(map[string]interface{}); ok {
+		dataMap["DestinationIdentity"] = safeToInt(dest["identity"])
+		dataMap["DestinationClusterName"] = extractString(dest, "cluster_name")
+		dataMap["DestinationNamespace"] = extractString(dest, "namespace")
+		dataMap["DestinationPodName"] = extractString(dest, "pod_name")
+		if workloads, ok := dest["workloads"].([]interface{}); ok {
+			dataMap["DestinationWorkloads"] = serializeToJSON(workloads)
+		}
+	}
+	// FlowType
+	dataMap["FlowType"] = extractString(flow, "Type")
+	// NodeName
+	dataMap["NodeName"] = extractString(flow, "node_name")
+	// Layer7
+	if l7, ok := flow["l7"].(map[string]interface{}); ok {
+		dataMap["Layer7"] = serializeToJSON(l7)
+	}
+	// Reply
+	if isReply, ok := flow["is_reply"].(bool); ok {
+		dataMap["Reply"] = isReply
+	}
+	// EventType
+	if eventType, ok := flow["event_type"].(map[string]interface{}); ok {
+		dataMap["EventType"] = serializeToJSON(eventType)
+	}
+	// Service
+	serviceData := map[string]interface{}{
+		"SourceService":               extractString(flow, "source_service.name"),
+		"SourceServiceNamespace":      extractString(flow, "source_service.namespace"),
+		"DestinationService":          extractString(flow, "destination_service.name"),
+		"DestinationServiceNamespace": extractString(flow, "destination_service.namespace"),
+	}
+	dataMap["Service"] = serializeToJSON(serviceData)
+	// TrafficDirection and TraceObservationPoint
+	dataMap["TrafficDirection"] = extractString(flow, "traffic_direction")
+	dataMap["TraceObservationPoint"] = extractString(flow, "trace_observation_point")
+
+	// aggregation support needed
+	// // FlowState
+	// dataMap["FlowState"] = extractString(flow, "flow_state")
```

Comment:
remove commented code if its not required.

Reply:
Those are in design doc TBD fields